src/main/java/com/onelogin/sdk/conn/Client.java

```diff
@@ -4014,6 +4431,27 @@ private String getIdsListBatch(List<Long> resourceList, ExtractionContext contex
 		return strList;
 	}
 
+	private List<HashMap<String, String>> retrieveListNameValue(URIBuilder url, int versionId) throws OAuthSystemException, OAuthProblemException {
+		List<HashMap<String, String>> listNameValue = new ArrayList<HashMap<String, String>>();
+
+		OneloginOAuth2JSONResourceResponse oAuth2Response = (OneloginOAuth2JSONResourceResponse) executeCall(url, OAuth.HttpMethod.GET, OneloginOAuthJSONResourceResponse.class);
+		if (oAuth2Response.getResponseCode() == 200) {
+			HashMap<String, String> info;
+			JSONArray dataArray = oAuth2Response.getJSONArrayFromContent();
+			for(int i = 0; i < dataArray.length(); i++) {
+				JSONObject data = dataArray.getJSONObject(i);
+				info = new HashMap<String, String>();
+				info.put("name", data.optString("name"));
+				info.put("value", data.optString("value"));
+				listNameValue.add(info);
+			}
+		} else {
+			setResponseError(oAuth2Response, false);
+		}
```

Comment:
From the error I was getting and what I can tell in the code `OneloginOAuthJSONResourceResponse` can't handle arrays, but `OneloginOAuth2JSONResourceResponse` seems to work instead. Some condition and action values can return an empty body if it can be random text, so dataArray can sometimes be null.

```suggestion
		OneloginOAuth2JSONResourceResponse oAuth2Response = (OneloginOAuth2JSONResourceResponse) executeCall(url, OAuth.HttpMethod.GET, OneloginOAuth2JSONResourceResponse.class);
		if (oAuth2Response.getResponseCode() == 200) {
			HashMap<String, String> info;
			JSONArray dataArray = oAuth2Response.getJSONArrayFromContent();
			if (dataArray == null) return null;
			
			for(int i = 0; i < dataArray.length(); i++) {
				JSONObject data = dataArray.getJSONObject(i);
				info = new HashMap<String, String>();
				info.put("name", data.optString("name"));
				info.put("value", data.optString("value"));
				listNameValue.add(info);
			}
		} else {
			setResponseError(oAuth2Response, false);
		}
```

Reply:
:+1: